### ISSUE TITLE
NsClient 0.5.0.62 install config

### ISSUE THREAD

**Issue by slythefox** [mickem/nscp#455]
Hi,
I am trying to customize a quiet install of NSClient++ 0.5.0.62 by using a transform file, but the documentation on how to disable some features is scarce or inaccurate. For example if I do not want to install the "OP5Montoring", "Documentation", and "SampleConfigs", and pass those as comma-separated values to the REMOVE parameter as per the Windows installer documentation, the installer log will say that it successfully installed the program, however the NSClient++ folder does not get created under %programfiles%, no service is installed, and it is not found in the installed programs. Furthermore, when saving a transformed MSI and testing a Custom Install, I still see the OP5 Monitoring feature selected and everything else but the MainProgram turned off, so REMOVE does not work as expected.
I also want to leave the autogenerated password blank, but "" will not work as a value. I want to use the LEGACY mode, but it does not work either once I install. I have been banging my head on it for 4 hours straight and I'm on my wits end. A little help would be appreciated.

These are the only salient docs I have found on the matter, but they did not help much:

http://docs.nsclient.org/manual/windows_installer.html#features

https://github.com/mickem/nscp/issues/386

Best,

RFR

**Comment by mickem** [comment]
If I am not mistaken unless you specify ADDDEFAULT=ALL no components wil lbe installed. Also the order of the various feature options are important (this is all per msiexec)

Now I am not a msi expert so I usually have to play around a bit before I get it right...

**Comment by mickem** [comment]
PS. Please also note that for nsclient++ options 0.5.1 has been greatly improved, but that is not relevant when it comes to feature selection (just things like PASSWORD and such)

**Comment by slythefox** [comment]
Understood! Thanks a lot! I will keep poking around and will post when I
get it right

**Comment by mickem** [comment]
FIY apart from OP5Montoring (notice the missing i) being misspelled (in the installer) it seems to work exactly as in the docs:

![image](https://user-images.githubusercontent.com/1434697/29493529-97bfd986-8598-11e7-8786-6738753ad118.png)

**Comment by slythefox** [comment]
Yes! That's it! Thanks for the tips, I was able to achieve all of my objectives.

The example I found  in the Wiki was:
msiexec /qn /l* log.txt /i NSCP-0.4.3.50-x64.msi INSTALLLOCATION=c:\FooBar CONFIGURATION_TYPE=registry://HKEY_LOCAL_MACHINE/software/NSClient++ ADDDEFAULT=ALL REMOVE=PythonScript

The ADDDEFAULT + REMOVE param combination does not seem to have any effect on v.0.5.0.62-x64, but ADDLOCAL with REMOVE works!

So to recap, the objectives were:
* Remove OP5 Monitoring, Documentation, and Sample Config
* Set to (insecure) legacy mode
* Leave the password field empty
* Enable common check plugins
* Enable Nrpe server

I put the following properties in a transform (mst):
MONITORING_TOOL=None
ALLOWED_HOSTS=nagios1.mydomcin.com,nagios2.mydomain.com
NRPEMODE=LEGACY
ADDLOCAL=ALL
REMOVE=OP5Montoring,Documentation,SampleConfig
CONF_CAN_CHANGE=1
CONF_CHECKS=1
CONF_NRPE=1

NSCLIENT_PWD="" had to be passed from the command line in order to be blank, because it would print the double quotes if specified in the transform.

So msiexec /i NSCP-0.5.0.62-x64.msi TRANSFORMS=MyTransforms.mst NSCLIENT_PWD="" now works

And saving the msi with the transforms applied, actually cuts install time dramatically.

Thank you very much for your help @mickem
